### DISCUSSION TITLE
Cannot Download anything with Version 4.2

### DISCUSSION THREAD

**Discussion by kishan167** [streamlink/streamlink#4665] [Q&A] [answered]
Hi there
before on my old version of Streamlink i can download most of the videos but after update i cannot download anything from Hotstar

**Downloading from**
https://www.hotstar.com/in/tv/bigg-boss-jodigal/1260061550/love-battles/1100057066

**Link given** Proxy Not added due to Privacy 
https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658203787~exp=1658204387~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=AOnmpIfBbrIKOKSP1kJRSMS9R8KnnQ7q4pCWLUcSm70F-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=0abbb7a95a1fd5bc3133bb568ad5e1964f86bff8cee68f199577f596433a5f71

**Error**
error: Unable to open URL: https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658203787~exp=1658204387~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=AOnmpIfBbrIKOKSP1kJRSMS9R8KnnQ7q4pCWLUcSm70F-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=0abbb7a95a1fd5bc3133bb568ad5e1964f86bff8cee68f199577f596433a5f71 (403 Client Error: Forbidden for url: https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658203787~exp=1658204387~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=AOnmpIfBbrIKOKSP1kJRSMS9R8KnnQ7q4pCWLUcSm70F-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=0abbb7a95a1fd5bc3133bb568ad5e1964f86bff8cee68f199577f596433a5f71)

**Full Cmd Used**
streamlink --http-proxy "" -o "master-859aabfa4b.ts" "https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057455/v3/1643796796827/184d287bc9fb331d44df77fbd5cbd155/master-859aabfa4b.m3u8?hdnea=st=1658204003~exp=1658204603~acl=/videos/hotstarint/1260061550/1100057455/v3/1643796796827/184d287bc9fb331d44df77fbd5cbd155/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=AOnmpIfBbrIKOKSP1kJRSMS9R8KnnQ7q4pCWLUcSm70F-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=2d5344e49451f7c0aa53a70b8dbb05a6cc76dcf5aeaa7780e2d80451694538d9" best

**Comment by kishan167** [comment, marked as answer]
do you guys know why its saying this Error now

After Reinstalling 
**URL:** https://www.hotstar.com/in/tv/bigg-boss-jodigal/1260061550/love-battles/1100057066

C:\Users\BosS>streamlink -o "D:\Downloads\Bigg Boss JodigalS01EP0.ts" --http-proxy "http://142.93.217.61:6060" --http-header "User-Agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:102.0) Gecko/20100101 Firefox/102.0" --http-header "Cookie=hs_uid=9acd7dd1-e4c6-4355-a94b-687b28fef333; ajs_user_id=%22d5139e7644434bbc98c174f8e250768b%22; ajs_group_id=null; ajs_anonymous_id=%22b2c0dabc-337c-4d7f-9ad3-1d0d4e32732e%22; _gcl_au=1.1.1661454995.1658201554; _ga_QV5FD29XJC=GS1.1.1658220631.2.1.1658225536.60; _ga=GA1.2.428367852.1658201555; _fbp=fb.1.1658201558664.2111183921; _gid=GA1.2.67967565.1658201559; _clck=11piuvm|1|f3a|0; _clsk=1mhb1km|1658225512503|65|0|h.clarity.ms/collect; hdntl=exp=1658288608~acl=%2f*~data=hdntl~hmac=3e52ea87225a84c18a8b7265307bf6f17b1d2a886bfde908720642a02bc0d9b2; _uetsid=6e3ddec0071311ed9f50212cce0ed0c7; _uetvid=6e3dde70071311eda0409d0a6d59fdc6" --http-header "Referer=https://www.hotstar.com/" "https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658226484~exp=1658227084~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=8Am3CB9ouZWzlhNkAREjsXm6PpwUrRZAcXW9zSziNS73-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=7cdb98a3a44170be7a531df9d6d892cc5ad4ec703db44f3c1a3a572b16ca09a5" best
usage: streamlink [OPTIONS] <URL> [STREAM]
streamlink: error: unrecognized arguments: https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658226484~exp=1658227084~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=8Am3CB9ouZWzlhNkAREjsXm6PpwUrRZAcXW9zSziNS73-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=7cdb98a3a44170be7a531df9d6d892cc5ad4ec703db44f3c1a3a572b16ca09a5 best

**Reply by bastimeyer** [reply]
Fix your broken Streamlink install. As mentioned in #4666, the debug log output was not one printed by a real Streamlink 4.2.0 installation.

According to what you've posted here, the command should output
```
[cli][info] Found matching plugin hls for URL https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658226484~exp=1658227084~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=8Am3CB9ouZWzlhNkAREjsXm6PpwUrRZAcXW9zSziNS73-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=7cdb98a3a44170be7a531df9d6d892cc5ad4ec703db44f3c1a3a572b16ca09a5
error: Unable to open URL: https://hses5.hotstar.com/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658226484~exp=1658227084~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=8Am3CB9ouZWzlhNkAREjsXm6PpwUrRZAcXW9zSziNS73-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=7cdb98a3a44170be7a531df9d6d892cc5ad4ec703db44f3c1a3a572b16ca09a5 (HTTPSConnectionPool(host='hses5.hotstar.com', port=443): Max retries exceeded with url: /videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/master-d60a40979b-phone.m3u8?hdnea=st=1658226484~exp=1658227084~acl=/videos/hotstarint/1260061550/1100057066/1625927003931/f28514bccb7b266312a8c3385f1503ae/*~data=ip=lkvxvRomM5KqojaFIn2ojw-userid=8Am3CB9ouZWzlhNkAREjsXm6PpwUrRZAcXW9zSziNS73-did=OJb3bpx2wNqJ6lvWjQMKFkZtcpO4AcsPT04VkZyFYMFAAUA0orcBzZI-cc=in-~hmac=7cdb98a3a44170be7a531df9d6d892cc5ad4ec703db44f3c1a3a572b16ca09a5 (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required'))))
```

However, as I also already said, custom HLS stream URLs are unsupported, and you'll have to figure it out on your own.

**Reply by kishan167** [reply]
i reinstalled it but can you share your cmd pls as you getting 407 Error

**Reply by bastimeyer** [reply]
I simply executed the command you've posted. Since this isn't an issue caused by shell tokenization, aka. missing quotation or special character escaping, this suggests that your install is broken, especially since your debug log output posted in #4666 is completely different to what Streamlink 4.2.0 would output.

"reinstalling" doesn't mean fixing. You need to **fix** your streamlink install. Completely remove streamlink from your system and install it again. Depending on how you installed it, since you're a Windows user, you will have to take a look at different install locations.

That's all I can tell you. The authentication and stream access issue can't be solved by anyone other than you. Good luck.
